src/Lean/Parser/Tactic.lean

```diff
@@ -66,13 +66,60 @@ doing a pattern match. This is equivalent to `fun` with match arms in term mode.
 @[builtin_tactic_parser] def introMatch := leading_parser
   nonReservedSymbol "intro" >> matchAlts
 
-/-- `decide` will attempt to prove a goal of type `p` by synthesizing an instance
-of `Decidable p` and then evaluating it to `isTrue ..`. Because this uses kernel
-computation to evaluate the term, it may not work in the presence of definitions
-by well founded recursion, since this requires reducing proofs.
-```
+/--
+`decide` attempts to prove the main goal (with target type `p`) by synthesizing an instance of `Decidable p`
+and then reducing that instance to evaluate the truth value of `p`.
+If it reduces to `isTrue h`, then `h` is a proof of `p` that closes the goal.
+
+Limitations:
+- The target is not allowed to contain local variables or metavariables.
+  If there are local variables, you can try first using the `revert` tactic with these local variables
+  to move them into the target, which may allow `decide` to succeed.
+- Because this uses kernel reduction to evaluate the term, `Decidable` instances defined
+  by well-founded recursion might not work, because evaluating them requires reducing proofs.
+  The kernel can also get stuck reducing `Decidable` instances with `Eq.rec` terms for rewriting propositions.
+  These can appear for instances defined using the `rw` and `simp` tactics for example --
+  make sure to use definitions such as `decidable_of_iff` instead.
```

Comment:
```suggestion
  These can appear for instances defined using the `rw` and `simp` tactics, for example, so
  make sure to use definitions such as `decidable_of_iff` instead.
```

I think this is a bit nicer style.

But in any case, a dash here should be an em dash, which should be done via Unicode because Markdown doesn't support TeX-style dash rewriting, and even if it did, `--` is an en dash.

Reply:
Right, `--` is commonly used as a substitute for an em dash, and if this were TeX I would have used `---`. I'm happy to substitute a Unicode symbol.

I don't think using the conjunction "so" would be right here, since the command "make sure [...]" is a separate (yet related) thought, with "instead" referring to the example tactics as well as anything that introduces `Eq.rec`. The first part is not in itself the reason for the directive.

Reply:
I now am not sure where our communication is going wrong, but we're not understanding each other! That's a sign that the text could be simpler, I think. What about something like:
```suggestion
  These can appear for instances defined using tactics (such as `rw` and `simp`). To avoid this,
  use definitions such as `decidable_of_iff` instead.
```

Reply:
(and I've seen the TeX convention of `--` being an en dash in enough contexts that I really think the Unicode is the right solution here)

Reply:
I've seen the TeX convention used in documents that were plainly not intended to be TeX. Though not for some years.

I think we can conclude that double-dashing conventions are varied and confusing, highly dependent on where one came from, and so Unicode is the way forward :-)